### ISSUE TITLE
Parser EOF when using double single quotes for escpaing in pattern (postgres)

### ISSUE THREAD

**Issue by minikomi** [layerware/hugsql#89]
I'm building some strings for pattern matching from user input, and trying to escape single quotes within patterns. I think I hit an edge case while doing so:

```clj
(hugsql.parser/parse "-- :name test\nselect * from emp where col1 ~~* '%''%'")
```

Causes:

```
  Show: Project-Only All 
  Hide: Clojure Java REPL Tooling Duplicates  (16 frames hidden)

1. Unhandled clojure.lang.ExceptionInfo
   SQL String terminated unexpectedly with EOF line: 2, column: 38
   {:line 2, :column 38}

                  core.clj: 4739  clojure.core/ex-info
                  core.clj: 4739  clojure.core/ex-info
                parser.clj:   11  hugsql.parser/parse-error
                parser.clj:    5  hugsql.parser/parse-error
                parser.clj:    7  hugsql.parser/parse-error
                parser.clj:    5  hugsql.parser/parse-error
                parser.clj:  206  hugsql.parser/read-sql-quoted
                parser.clj:  200  hugsql.parser/read-sql-quoted
                  AFn.java:  156  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  Var.java:  702  clojure.lang.Var/applyTo
                  core.clj:  657  clojure.core/apply
                  core.clj:  652  clojure.core/apply
                 debug.clj:  502  cider.nrepl.middleware.debug/apply-instrumented-maybe
                 debug.clj:  498  cider.nrepl.middleware.debug/apply-instrumented-maybe
                      REPL:  290  hugsql.parser/eval56477/parse/fn
                      REPL:  245  hugsql.parser/eval56477/parse
                  AFn.java:  156  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  Var.java:  702  clojure.lang.Var/applyTo
                  core.clj:  657  clojure.core/apply
                  core.clj:  652  clojure.core/apply
                 debug.clj:  502  cider.nrepl.middleware.debug/apply-instrumented-maybe
                 debug.clj:  498  cider.nrepl.middleware.debug/apply-instrumented-maybe
                      REPL:  224  hugsql.parser/eval56477/parse
                      REPL: 6709  boot.user/eval57790
                      REPL: 6709  boot.user/eval57790
             Compiler.java: 7062  clojure.lang.Compiler/eval
             Compiler.java: 7025  clojure.lang.Compiler/eval
                  core.clj: 3206  clojure.core/eval
                  core.clj: 3202  clojure.core/eval
                  main.clj:  243  clojure.main/repl/read-eval-print/fn
                  main.clj:  243  clojure.main/repl/read-eval-print
                  main.clj:  261  clojure.main/repl/fn
                  main.clj:  261  clojure.main/repl
                  main.clj:  177  clojure.main/repl
               RestFn.java:  137  clojure.lang.RestFn/applyTo
                  core.clj:  657  clojure.core/apply
                  core.clj:  652  clojure.core/apply
                regrow.clj:   18  refactor-nrepl.ns.slam.hound.regrow/wrap-clojure-repl/fn
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  clojure.tools.nrepl.middleware.interruptible-eval/evaluate/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  657  clojure.core/apply
                  core.clj: 1965  clojure.core/with-bindings*
                  core.clj: 1965  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   85  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   55  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  222  clojure.tools.nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
    interruptible_eval.clj:  190  clojure.tools.nrepl.middleware.interruptible-eval/run-next/fn
                  AFn.java:   22  clojure.lang.AFn/run
   ThreadPoolExecutor.java: 1149  java.util.concurrent.ThreadPoolExecutor/runWorker
   ThreadPoolExecutor.java:  624  java.util.concurrent.ThreadPoolExecutor$Worker/run
               Thread.java:  748  java.lang.Thread/run
```

**Comment by minikomi** [comment]
Unfortunately, I'm trying to match against multiple space-separated strings provided by a user against multiple fields, and using clojure to build up the query like:

```clj
   (clojure.string/join
    " OR "
    (for [st (:search-terms params)
          field ["s.web_id" "s.title"]]
      (str field " ~~* '%" st "%'")))
```

`ILIKE` is fairly easy to escape -- only a few special characters.

**Comment by minikomi** [comment]
I ended up with:

``` clj
/*~ (when (not-empty (:search-terms params)) */
   AND (
     s.web_id ILIKE ANY (ARRAY[:v*:search-terms])
     OR
     s.title ILIKE ANY (ARRAY[:v*:search-terms])
   )
/*~ ) ~*/
```

And escaping `%`,  `\`, `_` in the search terms.

**Comment by csummers** [comment]
This is fixed with c555c7158932d4c128885e451d2ec314f15e8a5c and will ship in 0.5.0 soon.

**Comment by csummers** [comment]
This is released in `0.5.0`.
